### PR TITLE
Added Apache Module & Combatflite import (from original branch)

### PR DESCRIPTION
**Pull request by PieterL1976**

### CODE REVIEW

File: dcs-dtc/Models/AH64/Waypoints/Waypoint.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+
+namespace DTC.Models.AH64.Waypoints
+{
+    public class Waypoint
+    {
+        private static Regex MGRSRegex = new Regex("^(\\d\\d\\s[A-Z][A-Z][A-Z]\\s\\d\\d\\d\\d\\s\\d\\d\\d\\d)$"); //MGRS format like 40 RDR 5123 6456
```
Comment (p-louis) [inline dcs-dtc/Models/AH64/Waypoints/Waypoint.cs:9]: Spaces should be optional and in different positions. My suggestion would be 
```
"^(\\d{2}\\s*[A-Z]\\s*[A-Z]{2}\\s*\\d{4}\\s*\\d{4})$"
```
`40RDR51236456` is just as valid as `40 R DR 5123 6456` or `40R DR12345678`

File: dcs-dtc/Models/AH64/Upload/WaypointBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                AppendCommand(ku.GetCommand("ENTR"));
+                AppendCommand(BuildString(ku, wpt.Elevation.ToString()));
+                AppendCommand(ku.GetCommand("ENTR"));
+                AppendCommand(mpd.GetCommand("TSD"));
+            }
+            AppendCommand(mpd.GetCommand("TSD"));
+
+        }
+
+        private string BuildString(Device ku, string mgrs)
```
Comment (p-louis) [inline dcs-dtc/Models/AH64/Upload/WaypointBuilder.cs:81]: Maybe rename this to `BuildMgrsString` to better reflect it's purpose

File: dcs-dtc/Models/AH64/AH64Configuration.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                if (pos == null)
+                {
+                    continue;
+                }
+
+                var lat = pos.Element("Latitude")?.Value;
+                var lon = pos.Element("Longitude")?.Value;
+                var dLat = double.Parse(lat.Replace('.', ','));
+                var dLon = double.Parse(lon.Replace('.', ','));
+                //if (
```
Comment (p-louis) [inline dcs-dtc/Models/AH64/AH64Configuration.cs:104]: Please remove the commented lines

File: dcs-dtc/UI/Aircrafts/AH64/UploadToJetPage.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        }
+        private void CheckUploadButtonEnabled()
+        {
+            btnUpload.Enabled = (_cfg.Waypoints.EnableUpload || _cfg.Radios.EnableUpload);
+        }
+
+        public override string GetPageTitle()
+        {
+            return "Upload to Heli";
```
Comment (p-louis) [inline dcs-dtc/UI/Aircrafts/AH64/UploadToJetPage.cs:51]: Maybe the filename should also be changed to UploadToHeliPage or so

**Review by p-louis — CHANGES_REQUESTED** [review]
Some smaller changes plus merge-conflict resolution is required. Beyond that it looks good to me

File: dcs-dtc/UI/Aircrafts/AH64/WaypointEdit_old.cs [hunk 5]
```diff
@@ -0,0 +1,411 @@
+﻿using DTC.Models;
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/UI/Aircrafts/AH64/WaypointEdit_old.cs]: Done, file removed

File: dcs-dtc/UI/Aircrafts/AH64/WaypointEdit_old.resx [hunk 6]
```diff
@@ -0,0 +1,120 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/UI/Aircrafts/AH64/WaypointEdit_old.resx]: Done, file removed

File: dcs-dtc/Models/Presets/Aircraft.cs [hunk 7]
```diff
@@ -1,76 +1,114 @@
...
+			{
+				if (Model == AircraftModel.F16C)
+				{
+					return "F-16C";
+				}
+				else if (Model == AircraftModel.FA18C)
+				{
+					return "F/A-18C";
+				}
+                else if (Model == AircraftModel.AH64D)
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs:23]: Strange, i don't see it that way in Visual Studio or notepad++ locally.

File: dcs-dtc/Models/Presets/Aircraft.cs [hunk 8]
```diff
@@ -1,76 +1,114 @@
...
+		{
+			if (Model == AircraftModel.F16C)
+			{
+				return typeof(F16Configuration);
+			}
+			else if (Model == AircraftModel.FA18C)
+			{
+				return typeof(FA18Configuration);
+			}
+            else if (Model == AircraftModel.AH64D)
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs:55]: same as above

File: dcs-dtc/Models/Presets/Aircraft.cs [hunk 9]
```diff
@@ -1,76 +1,114 @@
...
+			{
+				if (cfg == null)
+				{
+					cfg = new FA18Configuration();
+				}
+				var p = new Preset(name, cfg);
+				Presets.Add(p);
+				return p;
+			}
+            else if (Model == AircraftModel.AH64D)
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs:84]: same as above

File: dcs-dtc/UI/Aircrafts/AH64/LoadSavePage.cs [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+            grpLoad.Visible = true;
+            grpSave.Visible = true;
+            DisableLoadControls();
+        }
+
+        private void optXML_CheckedChanged(object sender, EventArgs e)
+        {
+            _configToLoad = null;
+            grpLoad.Text = "Load from Combatflite XML";
+            //grpSave.Text = "Save to Clipboard";
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/UI/Aircrafts/AH64/LoadSavePage.cs]: done, removed

File: dcs-dtc/Models/F16/Waypoints/WaypointSystem.cs [hunk 11]
```diff
@@ -63,5 +63,9 @@ private void RecalculateSequence()
 				wpt.Sequence = i + 1;
 			}
 		}
-	}
+		public int Count()
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/F16/Waypoints/WaypointSystem.cs]: Method removed.

File: dcs-dtc/Models/F16/F16Upload.cs [hunk 12]
```diff
@@ -21,8 +21,9 @@ public F16Upload(F16Configuration cfg)
 
 			_cfg = cfg;
 		}
+        internal F16Configuration Cfg => _cfg;
 
-		public void Load()
+        public void Load()
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/F16/F16Upload.cs:26]: same as above

File: dcs-dtc/Models/Base/FileStorage.cs [hunk 13]
```diff
@@ -26,7 +27,13 @@ private static string GetAirbasesFilePath()
 			return Path.Combine(path, "dtc-airbases.json");
 		}
 
-		private static string GetEmittersFilePath()
+        private static string GetIdentsFilePath()
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Base/FileStorage.cs:30]: same as above

File: dcs-dtc/Models/AH64/temp [hunk 14]
```diff
@@ -0,0 +1 @@
+vnbvbn
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/temp]: File removed

File: dcs-dtc/Models/AH64/AH64Commands.cs [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+namespace DTC.Models.AH64
+{
+    public class AH64Commands : IAircraftDeviceManager
+    {
+        private Dictionary<string, Device> Devices = new Dictionary<string, Device>();
+        public AH64Commands()
+        {
+            var delay = Settings.CommandDelayMs;
+
+            var cpgrmpd = new Device(47, "CPG_Right_MPD");
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/AH64Commands.cs:14]: Works from both cockpit seats, changed to ku & mpd.

File: dcs-dtc/Models/AH64/Waypoints/Waypoint.cs (hunk 1 above)
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/Waypoints/Waypoint.cs]: replaced regex

File: dcs-dtc/Models/AH64/Upload/WaypointBuilder.cs (hunk 2 above)
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/Upload/WaypointBuilder.cs]: Changed to BuildInputString, as it actually cleans the input for other fields as well.

File: dcs-dtc/UI/Aircrafts/AH64/UploadToJetPage.cs (hunk 4 above)
Comment (PieterL1976) [reply dcs-dtc/UI/Aircrafts/AH64/UploadToJetPage.cs]: Changed to UploadToHeliPage

File: dcs-dtc/Models/AH64/AH64Configuration.cs (hunk 3 above)
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/AH64Configuration.cs]: comments removed

File: dcs-dtc/Models/AH64/Waypoints/WaypointSystem.cs [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+        private void RecalculateSequence()
+        {
+            for (int i = 0; i < Waypoints.Count; i++)
+            {
+                Waypoint wpt = Waypoints[i];
+                wpt.Sequence = i + 1;
+            }
+        }
+
+        public int Count()
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/AH64/Waypoints/WaypointSystem.cs]: removed method

File: README.md [hunk 17]
```diff
@@ -20,8 +33,8 @@ This application is written using .NET Framework 4.7.2. You may want to download
 
 ## Installation
 
-- Grab the zip file from the Releases tab on Github.
-- Unzip it into a folder of your choice.
+- Grab the files from the Releases tab on Github.
```
[the remark this answers is not in the document]
Comment (PieterL1976) [reply README.md]: Indeed, changed

File: dcs-dtc/Models/Presets/Aircraft.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs]: Fixed on github site

File: dcs-dtc/Models/Presets/Aircraft.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs:55]: Fixed on github site

File: dcs-dtc/Models/Presets/Aircraft.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Presets/Aircraft.cs]: Fixed on github site

File: dcs-dtc/Models/F16/F16Upload.cs (hunk 12 above)
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/F16/F16Upload.cs]: Fixed on github site

File: dcs-dtc/Models/Base/FileStorage.cs (hunk 13 above)
[the remark this answers is not in the document]
Comment (PieterL1976) [reply dcs-dtc/Models/Base/FileStorage.cs]: Fixed on github site

**Comment by PieterL1976** [comment]
Should be OK now

**Review by p-louis — APPROVED** [review]
LGTM

**Comment by p-louis** [comment]
The failing build should be fixed, other than that I'm happy with the changes

**Pull request merged**